src/app/_components/MarkdownContent.tsx

```diff
@@ -16,6 +16,22 @@ export function MarkdownContent({ children }: MarkdownContentProps) {
         img: ({ src, alt }) => (
           <CustomImage src={src as string} alt={alt as string} />
         ),
+        figure: ({ node, ...props }) => (
+          <figure {...props} className="text-center">
+            {props.children}
+          </figure>
+        ),
+        figcaption: ({ node, ...props }) => (
+          <figcaption
+            {...props}
+            className="overflow-hidden overflow-ellipsis whitespace-nowrap break-words text-left font-sans text-[10px] font-thin"
+          >
+            {props.children}
+          </figcaption>
+        ),
+        iframe: ({ node, ...props }) => (
+          <iframe {...props} className="border-0" />
+        ),
```

Comment:
Why not add the style of these 3 elements in `tailwind.config.js`, to stay consistent with this existing setup?

In the case of `img` just above, we have to render `CustomImage` (a custom component), so we need this extra configuration. With `figure`, `figcaption` and `iframe`, we're only adding style to the existing HTML markup so we don't need to recreate it. Unless I'm missing something? Does that make sense?

Reply:
Reason why I included it is to make it self-contained, so the styles only apply when these elements are used in the markdown context. But since we already have `figcaption` in tailwind, it makes sense to move all of those there.  👍

Reply:
I believe the styles we define under `typography` in the `tailwind.config` are only applied when `className="prose"` is applied, which is the case on `MarkdownContent`. So we're safe on that front :)

https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#customizing-the-css